### PR TITLE
Keep openAtLoginSet state, use writeFileSync

### PR DESCRIPTION
**Pull request by patrickxb**
We have some GH reports of users who are removing the keybase login item via system preferences but then it comes back when they start the app.

I've repro'd this twice, and I think the writeFileSync change will fix those as they appear to be races writing the app-state.json file.  If that file gets hosed, it uses the defaults which will always install the login item.

The other weird thing is that if your display resolution changes, it considers app-state.json to be invalid, so the openAtLoginSet state was lost and the login item gets installed.

My other theory is that permissions on the file and/or directory are messed up and the app can't save the openAtLoginSet state.  Waiting to hear back from a user about that.  My guess is this is the issue because it sounds like it happens every time for the user.

cc @maxtaco

### CODE REVIEW

File: shared/desktop/app/app-state.js [hunk 1]
```diff
@@ -199,6 +201,9 @@ export default class AppState {
       const stateLoaded = JSON.parse(fs.readFileSync(configPath, {encoding: 'utf8'}))
       if (this._isValidState(stateLoaded)) {
         this.state = stateLoaded
+      } else {
```
Comment (chrisnojima) [inline shared/desktop/app/app-state.js]: we actually want all the rest of the state and really just ignore the window stuff so i'd argue the flow should be
```
if (!this._isValidState(stateLoaded)) {
      stateLoaded.x = null
      stateLoaded.y =  null
}

this.state = stateLoaded
```

Comment (patrickxb) [reply shared/desktop/app/app-state.js]: changed.

Comment (chrisnojima) [reply shared/desktop/app/app-state.js]: no i mean , we still want to load state, we just don't want to keep the x/y if the screen dimensions changed, but all other fields should be loaded

Comment (patrickxb) [reply shared/desktop/app/app-state.js]: See `dec5db9`

File: shared/desktop/app/app-state.js [hunk 2]
```diff
@@ -197,9 +199,13 @@ export default class AppState {
     }
     try {
       const stateLoaded = JSON.parse(fs.readFileSync(configPath, {encoding: 'utf8'}))
-      if (this._isValidState(stateLoaded)) {
-        this.state = stateLoaded
+
+      if (!this._isValidState(stateLoaded)) {
```
Comment (chrisnojima) [inline shared/desktop/app/app-state.js]: approved but maybe we should rename this like _isValidWindowState

**Pull request merged**